utils/pattern/src/common.rs

```diff
@@ -111,6 +111,10 @@ pub trait PatternBackend: crate::private::Sealed + 'static + core::fmt::Debug {
     /// Iterates over the pattern items in a store.
     #[doc(hidden)] // TODO(#4467): Should be internal
     fn iter_items(store: &Self::Store) -> Self::Iter<'_>;
+
+    /// Writes the placeholder into the pattern string, the inverse of `Self::PlaceholderKey::from_str`
+    #[cfg(feature = "alloc")]
+    fn escape_placeholder(s: &mut String, placeholder: Self::PlaceholderKey<'_>);
```

Comment:
SUggestion: why `&mut String` and not `core::fmt::Write`?

Reply:
Because it's infallible

Reply:
ok that's reasonable.